cobbler/cobbler_collections/manager.py

```diff
@@ -173,7 +170,7 @@ def deserialize(self) -> None:
                 self.__serializer.deserialize(collection=args[0], topological=args[1])  # type: ignore
             except Exception as error:
                 raise CX(
-                    f"serializer: error loading collection {collection.collection_type()}: {error}."
+                    f"serializer: error loading collection {args[0].collection_type()}: {error}."
```

Comment:
I have copied the error message from my local VSCode:

```
Argument of type "Menus | Distros | Repos | Profiles | Images | Systems" cannot be assigned to parameter "collection" of type "Collection[ITEM@deserialize]" in function "deserialize"
  Type "Menus | Distros | Repos | Profiles | Images | Systems" is incompatible with type "Collection[Distro]"
    "Images" is incompatible with "Collection[Distro]"
      Type parameter "ITEM@Collection" is invariant, but "Image" is not the same as "Distro"Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)
---
(variable) args: tuple[Menus, bool] | tuple[Distros, bool] | tuple[Repos, bool] | tuple[Profiles, bool] | tuple[Images, bool] | tuple[Systems, bool]
```

I don't understand why it doesn't understand that all of those types are inheriting from `BaseItem`.

Reply:
Ahhh I now understand. I think this is a covariance vs invariance issue. As the error message says, `ITEM` is invariant. As such we need a covariant type to allow doing what we are doing.

However I don't know how to achieve that since we do want a generic that describes that a certain specific collection contains items of a single specific type.

This is further shown in the following screenshot:
![grafik](https://github.com/user-attachments/assets/e1b9927c-b283-48c6-9414-126b8623e1e3)

Reply:
This will be fixed with a future push